From 0e6a460ee7052e83e8e59acca90815af71a93866 Mon Sep 17 00:00:00 2001
From: Martin Kinsey <martin.f.kinsey.civ@us.navy.mil>
Date: Thu, 13 Mar 2025 18:12:12 +0000
Subject: [PATCH 09/29] SDK Beam Example resolution ImGui sliders use int
 instead of float

**Description:** SDK Beam Example resolution values use int instead of float

**Notes:** noticed while using the example investigating rocky beam graphics. resolution is an int, making the slider a float isn't useful

**Testing Performed:** release, changed both values with the ImGui controls
---
 Examples/BeamTest/BeamTest.cpp | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Examples/BeamTest/BeamTest.cpp b/Examples/BeamTest/BeamTest.cpp
index 29ba7a46..fd98525b 100644
--- a/Examples/BeamTest/BeamTest.cpp
+++ b/Examples/BeamTest/BeamTest.cpp
@@ -177,14 +177,14 @@ public:
       }
 
       // Cap Resolution
-      float capRes = capRes_;
-      IMGUI_ADD_ROW(ImGui::SliderFloat, "Cap Res.", &capRes_, 1.f, 20.f, "%.3f", ImGuiSliderFlags_AlwaysClamp);
+      int capRes = capRes_;
+      IMGUI_ADD_ROW(ImGui::SliderInt, "Cap Res.", &capRes_, 1, 20, "%d", ImGuiSliderFlags_AlwaysClamp);
       if (capRes != capRes_)
         needUpdate = true;
 
       // Cone Resolution
-      float coneRes = coneRes_;
-      IMGUI_ADD_ROW(ImGui::SliderFloat, "Cone Res.", &coneRes_, 4.f, 40.f, "%.3f", ImGuiSliderFlags_AlwaysClamp);
+      int coneRes = coneRes_;
+      IMGUI_ADD_ROW(ImGui::SliderInt, "Cone Res.", &coneRes_, 4, 40, "%d", ImGuiSliderFlags_AlwaysClamp);
       if (coneRes != coneRes_)
         needUpdate = true;
 
@@ -250,8 +250,8 @@ private:
     prefs->set_shaded(shaded_);
     prefs->set_blended(blended_);
     prefs->set_rendercone(renderCone_);
-    prefs->set_capresolution(capRes_);
-    prefs->set_coneresolution(coneRes_);
+    prefs->set_capresolution(static_cast<unsigned int>(capRes_));
+    prefs->set_coneresolution(static_cast<unsigned int>(coneRes_));
     prefs->set_animate(animate_);
     prefs->set_pulserate(0.1);
     prefs->set_pulsestipple(0xfff0);
@@ -285,8 +285,8 @@ private:
   float vertSize_ = 45.f;
   float azimuth_ = 0.f;
   float elevation_ = 0.f;
-  float capRes_ = 15.f;
-  float coneRes_ = 30.f;
+  int capRes_ = 15;
+  int coneRes_ = 30;
   bool useOffset_ = false;
   bool shaded_ = false;
   bool blended_ = true;
-- 
2.47.1.windows.2

